File: JAVA/src/arrayMedium/leaderInArray.java
package arrayMedium;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class leaderInArray {
    public static List<Integer> function(int[] arr){
        List<Integer> res = new ArrayList<>();
        int max = arr[arr.length-1];
        res.add(max);
        for (int i = arr.length-2; i >=0; i--) {
            if (arr[i] > max){
                 res.add(arr[i]);
                 max = arr[i];
            }
        }
        Collections.reverse(res);
        return res;
    }
    public static void main(String[] args) {
        int[] arr = {10, 22, 12, 3, 0, 6};
        List<Integer> result = function(arr);
        System.out.println(result);
    }
}

File: PYTHON/01/arrays_medium_problems/leader_in_array.py
class Solution:
    def function(self,arr):
        res = []
        n = len(arr)
        max = arr[n-1]
        res.append(max)
        for i in range(n-2,0,-1):
            if arr[i] > max:
                res.append(arr[i])
                max = arr[i]
        res.reverse()
        return res

if __name__ == '__main__':
    arr = [10, 22, 12, 3, 0, 6]
    obj = Solution()
    result = obj.function(arr)
    print(result)